packages/block-library/src/cover/edit/index.js

```diff
@@ -319,10 +332,81 @@ function CoverEdit( {
 		createErrorNotice( message, { type: 'snackbar' } );
 	};
 
+	const onSelectEmbedUrl = ( embedUrl, provider ) => {
+		// Set initial attributes with URL and provider
+		setAttributes( {
+			url: embedUrl,
+			backgroundType: EMBED_VIDEO_BACKGROUND_TYPE,
+			embedProvider: provider,
+			id: undefined,
+			focalPoint: undefined,
+			hasParallax: undefined,
+			isRepeated: undefined,
+			useFeaturedImage: undefined,
```

Comment:
These undefined properties make me wonder if we should do the opposite, when you pick a different backgroundType (other than embed) we should also nullify embedProvider and embedSrc